src/devices/uFireIse/UFireIse.cs

```diff
@@ -298,33 +298,35 @@ private void SendCommand(byte data)
 
         private float ReadRegister(byte register)
         {
-            byte[] data = { 0, 0, 0, 0 };
-
             ChangeRegister(register);
 
-            data[0] = _device.ReadByte();
-            data[1] = _device.ReadByte();
-            data[2] = _device.ReadByte();
-            data[3] = _device.ReadByte();
-
-            return Convert.ToSingle(RoundTotalDigits(BitConverter.ToSingle(data, 0), 7));
+            Span<byte> data = stackalloc byte[4]
+            {
```

Comment:
You don't need to initialize as you will read the data right after. So you can get rid of this part from line 304 to line 309.